tests/graphql/library_system/library_system_permissions/objects.py

```diff
@@ -0,0 +1,76 @@
+from simple_api.adapters.graphql.graphql import GraphQLAdapter
+from simple_api.adapters.utils import generate
+from simple_api.django_object.django_object import DjangoObject
+from simple_api.django_object.permissions import IsAuthenticated, DjangoPermission
+from simple_api.django_object.actions import CreateAction, UpdateAction, DeleteAction, DetailAction, ListAction, \
+    ModelAction
+from simple_api.object.permissions import Or, And
+
+from tests.graphql.graphql_test_utils import build_patterns
+
+from .models import Book as BookModel, Subscription as SubscriptionModel
+from django.contrib.auth.models import User as UserModel
+
+
+class IsAdmin(IsAuthenticated):
+    def permission_statement(self, request, obj, **kwargs):
+        return request.user.is_staff or request.user.is_superuser
+
+
+class IsNotRestricted(DjangoPermission):
+    def permission_statement(self, request, obj, **kwargs):
+        return not obj.restricted
+
+    def error_message(self, **kwargs):
+        return "Restricted books cannot be accessed."
+
+
+def lend_book(request, params, **kwargs):
+    if kwargs["obj"].borrowed:
```

Comment:
wtf? `kwargs["obj"].borrowed = not kwargs["obj"].borrowed`

Reply:
Chtěl jsem explicitně že to to má dělat, když to nedává smysl automaticky, že půjčování i vrací knihy, ale jako asi true že lepší řešení je to dát do komentáře než takhle

Reply:
pripada mi to z kategorie `if(x==5) then return true else return false` misto `return x==5`